### PR TITLE
LPD-35048: Export lar file with a asset-list that contains a inexistent group can cause exception and fail the entire export process

### PR DESCRIPTION
**Pull request by gustavolimav**
Ticket: https://liferay.atlassian.net/browse/LPD-35048

**Steps to Reproduce:**

1. Create a new site named "site 1".
2. Create a second site named "site 2".
3. In "site 2", create a dynamic collection and link “site 1”.
4. Delete "site 1".
5. Export "site 2" as a .lar file.

**Expected Result:**

- The export works as expected without errors.

**Actual Result:**

- A null pointer exception occurs during the export process.

Knowing that, the fix on the PR is right and necessary in the scenario that we don't remove linked sites from the collections, but I am thinking if we should file a new ticket to implement a more complete solution that remove linked sites if they are deleted. What do you think?

### CODE REVIEW

**Comment by gustavolimav** [comment]
ci:test:search

**Comment by gustavolimav** [comment]
ci:test:sf

**Comment by gustavolimav** [comment]
ci:test:relevant

**Comment by gustavolimav** [comment]
ci:stop

**Comment by gustavolimav** [comment]
ci:test:relevant

**Comment by gustavolimav** [comment]
ci:test:search

**Comment by joshuacords** [comment]
Hi @gustavolimav,

Did you have a plan for testing this behavior?

**Comment by gustavolimav** [comment]
Hi @joshuacords, do you think we need more than this test https://github.com/liferay-search/liferay-portal/pull/1462/commits/d29dde26348a86b66d3f22c614a2d7048f723b4c?

**Comment by joshuacords** [comment]
Sorry - I was looking at a non-refreshed version of the PR - I now see the test

**Comment by joshuacords** [comment]
I want to note that when you try to view the collection on Site 2, an error is thrown on screen. If this is the actual use case, we want to take care of that issue as well.

File: modules/apps/asset/asset-list-service/src/test/java/com/liferay/asset/list/internal/exportimport/content/processor/AssetListEntryExportImportContentProcessorTest.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+
+import org.mockito.Mockito;
+
+/**
+ * @author Gustavo Lima
+ */
+public class AssetListEntryExportImportContentProcessorTest {
```
Comment (joshuacords) [inline modules/apps/asset/asset-list-service/src/test/java/com/liferay/asset/list/internal/exportimport/content/processor/AssetListEntryExportImportContentProcessorTest.java:25]: Test should go in asset-list-test.

File: modules/apps/asset/asset-list-service/src/test/java/com/liferay/asset/list/internal/exportimport/content/processor/AssetListEntryExportImportContentProcessorTest.java [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+		LiferayUnitTestRule.INSTANCE;
+
+	@Before
+	public void setUp() {
+		_assetListEntryExportImportContentProcessor =
+			new AssetListEntryExportImportContentProcessor();
+	}
+
+	@Test
+	public void testAddGroupMappingsElementWithInexistentGroup() {
```
Comment (joshuacords) [inline modules/apps/asset/asset-list-service/src/test/java/com/liferay/asset/list/internal/exportimport/content/processor/AssetListEntryExportImportContentProcessorTest.java:39]: I don't really like this test, it seems too narrow to check for real problems. Could you look into making this an integration (or playwright) test if necessary to more realistically test the use case.

**Review by joshuacords — CHANGES_REQUESTED** [review]

File: modules/apps/asset/asset-list-service/src/test/java/com/liferay/asset/list/internal/exportimport/content/processor/AssetListEntryExportImportContentProcessorTest.java (hunk 2 above)
Comment (BryanEngler) [reply modules/apps/asset/asset-list-service/src/test/java/com/liferay/asset/list/internal/exportimport/content/processor/AssetListEntryExportImportContentProcessorTest.java:39]: i think the unit test is probably fine, lets just add another assert: addElement() is called once when when group is not null

**Comment by gustavolimav** [comment]
Hey @joshuacords @BryanEngler, changes were made. Thanks for the review.

**Comment by gustavolimav** [comment]
ci:test:search

**Comment by gustavolimav** [comment]
ci:test:sf

**Comment by gustavolimav** [comment]
ci:test:relevant

**Comment by gustavolimav** [comment]
https://github.com/liferay-search/liferay-portal/pull/1478

**Pull request closed without merging**